sdk/tools/winesync/winesync.py

```diff
@@ -67,7 +67,7 @@ def create_or_checkout_wine_branch(self, wine_tag, wine_staging_tag):
             wine_branch_name += '-' + wine_staging_tag
 
         branch = self.wine_repo.lookup_branch(wine_branch_name)
-        if branch is None:
+        if not branch:
```

Comment:
I am unsure about these changes, iirc older (or newer) python had some problems with the `if not branch` variant

Reply:
Admittedly I haven't tested it but which version had issues? Lowest py version we can/could use (before the change) is 3.6 because of fstrings

Reply:
This is doing something else:
`is None` checks explicitly if the object is `None`, `not branch` checks that the thing evaluates to `False`. (so it could be an object still)

Reply:
lookup_branch will return an error if it doesn't find a branch with a certain name in that case the script won't execute anyways, only place where `not` would have different behaviour would be if branch was "" but this returns a class instance `Branch` so that can't happen
![image](https://github.com/reactos/reactos/assets/36992813/a81cbfe2-10c7-4aa1-995e-27b301daa689)

Reply:
> lookup_branch will return an error if it doesn't find a branch with a certain name in that case the script won't execute anyways, only place where `not` would have different behaviour would be if branch was "" but this returns a class instance `Branch` so that can't happen 

Not only `""` evaluates to false, `[]` does, and a number of other things.
In general this will call the `__nonzero__` function, and if that is not implemented on the object it tries `__len__`.